bin/predictions.py

```diff
@@ -196,25 +223,35 @@
         results = results_bag.persist(optimize_graph=False)
         wait(results)
 
-    if pickle_in_config:
-        pickle_path = config["output_options"]["pickle_path"]
-
-        if pickle_path != "None":
-            if (
-                "output_all_structures" in config["output_options"]
-                and config["output_options"]["output_all_structures"]
-            ):
-                adslab_atoms = adslab_atoms_bag.compute(optimize_graph=False)
-                df_results["adslab_atoms"] = adslab_atoms
-                df_results.to_pickle(pickle_path)
-            else:
-                # screen classes from custom packages
-                class_mask = (
-                    df_results.columns.to_series()
-                    .apply(lambda x: str(type(df_results[x].iloc[0])))
-                    .apply(lambda x: "catkit" in x or "ocp" in x or "ocdata" in x)
-                )
-                df_results[class_mask[~class_mask].index].to_pickle(pickle_path)
-
-        with open(config["output_options"]["config_path"], "w") as fhandle:
+    if config["output_options"]["pickle_final_output"]:
+        pickle_path = f"outputs/{run_id}/results_df.pkl"
+
+        if (
+            "output_all_structures" in config["output_options"]
+            and config["output_options"]["output_all_structures"]
+        ):
+            adslab_atoms = adslab_atoms_bag.compute(optimize_graph=False)
+            df_results["adslab_atoms"] = adslab_atoms
+            df_results.to_pickle(pickle_path)
+            if not inference:
+                num_adslabs = sum(df_results["adslab_atoms"].apply(len))
+                filtered_slabs = len(df_results)
+                num_inferred = 0
+        else:
+            # screen classes from custom packages
+            class_mask = (
+                df_results.columns.to_series()
+                .apply(lambda x: str(type(df_results[x].iloc[0])))
+                .apply(lambda x: "catkit" in x or "ocp" in x or "ocdata" in x)
+            )
+            df_results[class_mask[~class_mask].index].to_pickle(pickle_path)
+
+        with open(f"outputs/{run_id}/inputs_config.yml", "w") as fhandle:
             yaml.dump(config, fhandle)
+
+    # Make final updates to the sankey diagram and plot it
+    unfiltered_slabs = unfiltered_surface_bag.count().compute()
```

Comment:
@zulissi @KirbyBroderick I would like to change this before merging. It takes a while to count this bag, any ideas for how to improve?